src/pal/src/configure.cmake

```diff
@@ -73,8 +73,33 @@ check_function_exists(directio HAVE_DIRECTIO)
 check_function_exists(semget HAS_SYSV_SEMAPHORES)
 check_function_exists(pthread_mutex_init HAS_PTHREAD_MUTEXES)
 check_function_exists(ttrace HAVE_TTRACE)
+if(CMAKE_SYSTEM_NAME STREQUAL Linux)
```

Comment:
Without specifying the libraries, CMAKE says it fails because CMAKE tries to do LINK (ld) as well.
Therefore, if the behavior of CMAKE is same in non-Linux systems, we need to know the exact names of such systems in order to reuse the check_cxx_source_compiles for non-Linux systems, which I cannot test.